src/Action/CheckEmailAction.php

```diff
@@ -15,32 +15,107 @@
+                    __METHOD__,
+                    Pool::class,
+                    \get_class($templateRegistry)
+                ));
+            }
+            $this->adminPool = $templateRegistry;
 
-    public function __invoke(Request $request): Response
-    {
-        $username = $request->query->get('username');
+            if (!$tokenTtl instanceof TemplateRegistryInterface) {
+                throw new \TypeError(sprintf(
+                    'Argument 4 passed to %s() must be an instance of %s, %s given.',
+                    __METHOD__,
+                    TemplateRegistryInterface::class,
+                    \get_class($tokenTtl)
+                ));
+            }
+            $this->templateRegistry = $tokenTtl;
+
+            if (!is_int($tokenTtlDeprecated)) {
+                throw new \TypeError(sprintf(
+                    'Argument 5 passed to %s() must be type of %s, %s given.',
+                    __METHOD__,
+                    'integer',
+                    \gettype($tokenTtlDeprecated)
+                ));
+            }
+            $this->tokenTtl = $tokenTtlDeprecated;
 
-        if (null === $username) {
-            // the user does not come from the sendEmail action
-            return new RedirectResponse($this->urlGenerator->generate('sonata_user_admin_resetting_request'));
+            @trigger_error(sprintf(
+                'Passing an instance of %s as argument 2 to "%s()" is deprecated since sonata-project/user-bundle 5.x and will only accept an instance of %s in version 6.0.',
+                UrlGeneratorInterface::class,
+                __METHOD__,
+                Pool::class
+            ), \E_USER_DEPRECATED);
+        } else {
+            $this->adminPool = $adminPool;
+            if (!$templateRegistry instanceof TemplateRegistryInterface) {
+                throw new \TypeError(sprintf(
+                    'Argument 3 passed to %s() must be an instance of %s, %s given.',
+                    __METHOD__,
+                    TemplateRegistryInterface::class,
+                    \get_class($templateRegistry)
+                ));
+            }
+            $this->templateRegistry = $templateRegistry;
+
+            if (!is_int($tokenTtl)) {
+                throw new \TypeError(sprintf(
+                    'Argument 4 passed to %s() must be type of %s, %s given.',
+                    __METHOD__,
+                    'integer',
+                    \gettype($tokenTtl)
+                ));
+            }
+            $this->tokenTtl = $tokenTtl;
+
+            if (null !== $tokenTtlDeprecated) {
+                throw new \TypeError(sprintf(
+                    'Argument 5 passed to %s() must be %s, %s given.',
+                    __METHOD__,
+                    'NULL',
+                    \gettype($tokenTtlDeprecated)
+                ));
+            }
         }
+    }
 
+    public function __invoke(): Response
```

Comment:
For BC, the `Request $request` param probably needs to be optional.
(with Deprecate Warning)

Reply:
No I don't think so.

There won't be any error if you pass an extra param

Reply:
Ah, i misunderstand the errors there,

it was only Psalm/PHPStan complaining, not a real PHP error.